client/my-sites/theme/main.jsx

```diff
@@ -643,8 +659,10 @@ export default connect(
 			isJetpack: isJetpackSite( state, siteId ),
 			isPremium: isThemePremium( state, id ),
 			isPurchased: isPremiumThemeAvailable( state, id, siteId ),
+			hasUnlimitedPremiumThemes: config.isEnabled( 'jetpack/pijp' ) && hasFeature( state, siteId, FEATURE_UNLIMITED_PREMIUM_THEMES ),
```

Comment:
Oh, I think we should drop this line for the same reason as https://github.com/Automattic/wp-calypso/pull/13621/files#r123494284, and change R607 to 

```js
const needsJetpackUpgrade = isJetpack && isPremium && config.isEnabled( 'jetpack/pijp' ) && ! isPurchased;
 ```